src/app/sommerjobb/_utils/sommerjobbElasticSearchRequestBody.ts

```diff
@@ -0,0 +1,964 @@
+    }
+
+    const internationalObject: BoolFilter = {
+        bool: {},
+    };
+
+    if (international) {
+        internationalObject.bool.must_not = {
+            term: {
+                "locationList.country.keyword": "NORGE",
+            },
+        };
+    }
+
+    if (Array.isArray(countries) && countries.length > 0) {
+        internationalObject.bool.should = [
+            ...countries.map((c) => ({
+                term: {
+                    "locationList.country.keyword": c,
+                },
+            })),
+        ];
+    }
+
+    if (
+        Object.keys(internationalObject.bool).includes("must_not") ||
+        Object.keys(internationalObject.bool).includes("should")
+    ) {
+        filter.nested.query.bool?.should?.push(internationalObject);
+    }
+
+    return filter;
+}
+
+function filterJanzzOccupation(occupation: string[] | undefined) {
+    const filters: BoolFilter[] = [];
+    if (occupation && occupation.length > 0) {
+        const filter: BoolFilter = {
+            bool: {
+                should: [],
+            },
+        };
+        occupation.forEach((item) => {
+            filter.bool?.should?.push({
+                term: {
+                    category_styrk08_facet: item,
+                },
+            });
+            filter.bool?.should?.push({
+                term: {
+                    searchtags_facet: item,
+                },
+            });
+        });
+        filters.push(filter);
+    }
+    return filters;
+}
+
+function filterOccupation(occupationFirstLevels: string[] | undefined, occupationSecondLevels: string[] = []) {
+    return filterNestedFacets(
+        occupationFirstLevels,
+        occupationSecondLevels,
+        "occupationList.level1",
+        "occupationList.level2",
+        "occupationList",
+    );
+}
+
+function filterSector(sector: string[] | undefined) {
+    const filters: BoolFilter[] = [];
+    if (sector && sector.length > 0) {
+        const filter: BoolFilter = {
+            bool: {
+                should: [],
+            },
+        };
+        sector.forEach((item) => {
+            filter.bool?.should?.push({
+                term: {
+                    sector_facet: item,
+                },
+            });
+        });
+        filters.push(filter);
+    }
+    return filters;
+}
+
+const sommerjobbKeywords = ["Sommerjobb", "Sommervikar", "Sesongarbeid"];
+
+function mainQueryTemplateFunc(qAsArray: string[]): BoolFilter {
+    const sommerjobbScoringProfile = [
+        "category_name_no^2",
+        "title_no^2",
+        "keywords_no^2",
+        "searchtagsai_no^2",
+        "searchtags_no^2",
+        "adtext_no^0.1",
+        "employerdescription_no^0.1",
+    ];
+
+    const sommerjobbCategoryScoringProfile = [
+        "category_name_no^1",
+        "title_no^1",
+        "keywords_no^2",
+        "searchtagsai_no^3",
+        "searchtags_no^2",
+        "adtext_no^0.1",
+        "employerdescription_no^0.1",
+    ];
```

Comment:
Her er scoringen for sommerjobb, ene for "sommerjobb keywords" og en for "kategori keywords". De er forholdsvis like men fra min testing så pleier "sommerjobb keywords" å komme ofte i overskriften så den er litt mer boostet mens kategori keywords legger jeg litt mer vekt på AI keywords. Bare endre i vei men generelt for begge så har jeg legget mer vekt på searchtagsai, keywords og searchtags enn tradisjonelle søket.